### ISSUE TITLE
RedundantModifierRule

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#251]
Flags modifier that are redundant. A simple case for this would be to flag `public` in this case:

```kotlin
public class Something
```

This might already be enough as a first step. More cases could be added later to it.

[Checkstyle reference](http://checkstyle.sourceforge.net/config_modifier.html#RedundantModifier)

**Comment by schalkms** [comment]
Just a note
- the stdlib uses it
- public can be used to increase the visibility of a inherited protected method

reference:
https://discuss.kotlinlang.org/t/is-the-public-modifier-obsolete/1393

**Comment by vanniktech** [comment]
The public case was just an example. For the case that `public` will increase the visibility from `protected` obviously that part should not be flagged.

**Comment by arturbosch** [comment]
Also see OptionalUnit and OptionalSemicolon rules. Maybe a `MultiRule` which handles all cases could be cool.

**Comment by arturbosch** [comment]
Overriden members should not get flagged as they can change the visibility to public.

**Comment by Mauin** [comment]
We have a basic implementation of this with the `RedundantVisibilityModifierRule` in #470. Should we leave this open to improve the rule for more cases? Otherwise lets close this.
